internal/httpproxy/http_proxy_test.go

```diff
@@ -100,6 +101,16 @@ func TestProxyConn_Close(t *testing.T) {
 		assert.Fail(t, "Timer should have been stopped")
 	default:
 	}
+
+	// Ensure metrics are only measured once
+	_ = proxyConn.Close()
```

Comment:
Why call `Close()` again here, it's already closed? Also, below is doing:

```
	proxyConn.once.Do(func() {
 		called = true
 	})
 ```
 is testing that the `sync.Once` cannot be reused after `Close()` but would it be better to mock the `tracker` to ensure that `p.tracker.recordConnMetrics()` is truly only called once?

Reply:
> Why call Close() again here, it's already closed?

We want to check if the metrics are truly measured only once. I updated the test ([cc51195](https://github.com/Twingate/kubernetes-access-gateway/pull/71/commits/cc5119575f459700ed462b7e8d5fa24e121e5d88)) to assert the `connTotal` instead - it has a simpler setup vs mocking the `tracker`